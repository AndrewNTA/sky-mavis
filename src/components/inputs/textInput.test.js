import React from 'react';
import renderer from 'react-test-renderer';
import TextInput from './textInput';

test('TextInput should render the input type text without action icon', () => {
  const component = renderer.create(
    <TextInput />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('TextInput should render the data readonly', () => {
  const component = renderer.create(
    <TextInput
      readonly={true}
      defaultValue={'Hello'}
    />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('TextInput should render the input with action icon', () => {
  const ActionIcon = <div>close</div>;
  const component = renderer.create(
    <TextInput
      actionIcon={ActionIcon}
    />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('TextInput should render the input type password', () => {
  const component = renderer.create(
    <TextInput
      type={'password'}
    />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});