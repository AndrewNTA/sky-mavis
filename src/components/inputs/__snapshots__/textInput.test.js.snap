// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render the data readonly 1`] = `
<div
  className="r-text-input-container readonly"
>
  <div
    className="r-text-input-readonly"
  >
    Hello
  </div>
</div>
`;

exports[`TextInput should render the input type password 1`] = `
<div
  className="r-text-input-container false"
>
  <input
    className="r-text-input"
    type="password"
  />
</div>
`;

exports[`TextInput should render the input type text without action icon 1`] = `
<div
  className="r-text-input-container false"
>
  <input
    className="r-text-input"
    type="text"
  />
</div>
`;

exports[`TextInput should render the input with action icon 1`] = `
<div
  className="r-text-input-container false"
>
  <input
    className="r-text-input"
    type="text"
  />
  <div>
    close
  </div>
</div>
`;
